libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -159,7 +160,68 @@ internal object MapboxNavigationTelemetry :
         description: String,
         @FeedbackEvent.Source feedbackSource: String,
         screenshot: String?,
-        feedbackSubType: Array<String>?,
+        feedbackSubType: Array<String>,
+        appMetadata: AppMetadata?
+    ) {
+        createUserFeedback(
+            {
+                sendMetricEvent(it)
+            },
+            feedbackType,
+            description,
+            feedbackSource,
+            screenshot,
+            feedbackSubType,
+            appMetadata
+        )
+    }
+
+    fun cacheUserFeedback(
+        @FeedbackEvent.Type feedbackType: String,
+        description: String,
+        @FeedbackEvent.Source feedbackSource: String,
+        screenshot: String?,
+        feedbackSubType: Array<String>,
+        appMetadata: AppMetadata?
+    ) {
+        createUserFeedback(
+            {
+                feedbackEventCacheMap[it.feedbackId] = it
+            },
+            feedbackType,
+            description,
+            feedbackSource,
+            screenshot,
+            feedbackSubType,
+            appMetadata
+        )
+    }
+
+    fun getCachedUserFeedback() = feedbackEventCacheMap.map {
+        it.value.getCachedNavigationFeedbackEvent()
+    }
+
+    fun postCachedUserFeedback(cachedFeedbackEventList: List<CachedNavigationFeedbackEvent>) {
+        log("post user feedback")
+        val feedbackEventCache = LinkedHashMap(feedbackEventCacheMap)
+        feedbackEventCacheMap.clear()
+
+        cachedFeedbackEventList.forEach { cachedFeedback ->
+            metricsReporter.addEvent(
+                feedbackEventCache[cachedFeedback.feedbackId]?.apply {
+                    update(cachedFeedback)
+                } ?: return@forEach
+            )
```

Comment:
why  do we need to update old events with new from a `cachedFeedbackEventList `?
can we just post new events because they are already updated?

Reply:
when we post new events, we lose the location information. we only need the user's input of `feedbackSubType` and `description`, that's why I do this way.